### PR TITLE
Use 4.0.99 as as intermediate model to keep 4.1.0 for the Maven 4.x model

### PR DESCRIPTION
**Pull request by gnodet**
Fix model doc generation, change the intermediate model to use 4.0.99 and align the new model to 4.1.0

This requires https://github.com/apache/maven-integration-testing/pull/315 to pass ITs...

### CODE REVIEW

**Review by cstamas — APPROVED** [review]

File: api/maven-api-model/src/main/mdo/maven.mdo [hunk 1]
```diff
@@ -387,7 +388,7 @@
       </fields>
       <codeSegments>
         <codeSegment>
-          <version>4.0.0/4.1.0</version>
+          <version>4.0.0/4.0.99</version>
```
Comment (kwin) [inline api/maven-api-model/src/main/mdo/maven.mdo:391]: @gnodet What is this supposed to mean?

Comment (gnodet) [reply api/maven-api-model/src/main/mdo/maven.mdo:391]: This is closed ranged version notation.  That's defined by modello...

Comment (kwin) [reply api/maven-api-model/src/main/mdo/maven.mdo:391]: Ah, sorry. I get it now. I thought this element was for documentation purposes

Comment (cstamas) [reply api/maven-api-model/src/main/mdo/maven.mdo:391]: Sadly, modello "version range" predates (AFAIK, someone may correct me) Maven ranges, and has totally different  syntax...

**Pull request merged**